### ISSUE TITLE
Octave pkg support?

### ISSUE THREAD

**Issue by apjanke** [bids-standard/bids-matlab#26]
Are you interested in rearranging the repo so that it could be installed using Octave's [`pkg` tool](https://octave.sourceforge.io/octave/function/pkg.html)? That's Octave's standard package manager. It provides conveniences like install/uninstall commands, a load command, and the ability to specify dependencies.

This would be a [simple rearrangement](https://octave.org/doc/v4.4.0/Creating-Packages.html): 
* stick the `+bids` source code directory in a new top-level `inst/` directory
* add a `DESCRIPTION` file in Octave `pkg` format
* copy or rename `LICENSE` to `COPYING`, to follow GNU's conventions
* optionally, add a `CITATION` file to describe how people should cite your package for academic publications

I could throw together a PR to do this, if you're interested.

**Comment by Remi-Gau** [comment]
I personally have nothing against it.

**Comment by robertoostenveld** [comment]
the consequence for naive MATLAB and Octave users (which I suspect to be the majority) would be that when downloading the master.zip file, they would have to do
```
addpath bids-matlab-master/inst
```
instead of
```
addpath bids-matlab-master
```

Would it be possible to make the octave package specifically at the release moment and keep the  directory structure of the repo as it is now?

**Comment by apjanke** [comment]
> Would it be possible to make the octave package specifically at the release moment and keep the directory structure of the repo as it is now?

Yes. You could just have a distribution-building step that copied the files into a temporary "target" directory, moving `+bids/` (and maybe `tests/`) into an `inst/` subfolder under that, and zip that up as your Octave pkg distribution.

**Comment by apjanke** [comment]
Ah, you know what? Then we'd end up with zip/tarball distribution files for the releases whose file layout didn't match the layout in the original source repo, so the installation instructions would have to be different for installing from those versus from a source checkout or GitHub main repo archive download. (E.g. we'd have to say "if you're running Matlab, *don't* use the `bids-matlab-0.1.0.zip` download; instead use `bids-matlab-master.zip`" or something like that.)

I'm now thinking this is maybe more trouble than it's worth unless there's actual user demand to have an Octave pkg package.
